app/models/offer.rb

```diff
@@ -0,0 +1,5 @@
+class Offer < ApplicationRecord
+  belongs_to :request
+  belongs_to :helper, class_name: 'User'
+  validates_uniqueness_of :helper_id, scope: :request_id, message: 'is already registered with this request'
```

Comment:
Is this actually working for you? I can't see how this might check that there's only one unique helper on a specific request.

Reply:
This worked. Also added the equivalent in database in case race conditions might show up one day in the future.